packages/core/src/browser/shell/application-shell.ts

```diff
@@ -498,9 +504,68 @@ export class ApplicationShell extends Widget {
         dockPanel.id = MAIN_AREA_ID;
         dockPanel.widgetAdded.connect((_, widget) => this.fireDidAddWidget(widget));
         dockPanel.widgetRemoved.connect((_, widget) => this.fireDidRemoveWidget(widget));
+
+        const openUri = async (fileUri: URI) => {
+            try {
+                const opener = await this.openerService.getOpener(fileUri);
+                opener.open(fileUri);
+            } catch (e) {
+                console.info(`no opener found for '${fileUri}'`);
+            }
+        };
+
+        dockPanel.node.addEventListener('drop', event => {
+            if (event.dataTransfer) {
+                const uris = this.additionalDraggedUris || ApplicationShell.getDraggedEditorUris(event.dataTransfer);
+                if (uris.length > 0) {
+                    uris.forEach(openUri);
+                } else if (event.dataTransfer.files?.length > 0) {
+                    // the files were dragged from the outside the workspace
+                    Array.from(event.dataTransfer.files).forEach(async file => {
```

Comment:
Minor: Any reason to make this function `async` if we don't await anything in there?

Reply:
It still should compile to a promise though and thus not block the main loop, right?

Reply:
It doesn't matter whether it's a promise or not when it comes to blocking the event loop. It will still be blocking, except if it calls an explicitly non-blocking function (such as `fs.promises`, `setTimeout` or `fetch`). As an example execute the following in any JS shell:

```
[0, 1, 2, 3].forEach(async (e) => { console.log(e) });
console.log('end')
```

It will print `0 1 2 3 end` as the `forEach` callback is instantly executed and blocking the event loop.

Reply:
So this doesn't compile to 
```
[0, 1, 2, 3].forEach(async (e) => { new Promise((resolve => {
   console.log(e);
   resolve();
})});
```
?

Reply:
My snippet is pure JS, no compilation step needed.

Reply:
Under the hood, it must convert to a promise, but I guess it does not push the leading code onto the task queue. So more likely it looks like:
```javascript
async (e) => {
   console.log(e);
   return Promise.resolve();
}
```

Reply:
Wat?!

Reply:
Anyway, having the `async` in there does not yield any benefits, as it doesn't affect the execution order. JS runtimes will always try to execute as much code as possible until they hit a non-blocking, awaited function. Enough argument to remove the `async`?